src/gt4py/gtscript.py

```diff
@@ -329,6 +329,35 @@ def _decorator(func):
     return _decorator(definition)
 
 
+class _AxisSplitter:
+    def __init__(self, axis: str, offset: int):
+        self.axis = axis
+        self.offset = offset
+
+    def __repr__(self):
+        return f"_AxisSplitter(axis={self.axis}, offset={self.offset})"
+
+    def __str__(self):
+        return f"{self.axis}[{self.offset}]"
+
+
+class _AxisInterval:
+    def __init__(self, axis: str, start: int, end: int):
+        assert start < end
```

Comment:
Wouldn't this assert break when using negative numbers indexing from then end of the interval like [1:-1]?

Reply:
Yes! Good catch.

Reply:
If this is the case, one of the tests was wrong or incomplete because it was explicitly testing an `[1:-1]` interval

Reply:
Ah, it never actually makes the AxisInterval in that case -- it only parses the AST as if it were one. If the `K[1:-1]` were a named external, it would have caught the code error.